File: backend/internal/models/log.go
package models

import (
	"time"
)

type Log struct {
	ID         string                 `json:"id"`
	Timestamp  time.Time              `json:"timestamp"`
	Level      string                 `json:"level"`
	Message    string                 `json:"message"`
	Service    string                 `json:"service"`
	TraceID    string                 `json:"trace_id,omitempty"`
	SpanID     string                 `json:"span_id,omitempty"`
	Attributes map[string]interface{} `json:"attributes,omitempty"`
}

type LogQuery struct {
	StartTime time.Time `json:"start_time"`
	EndTime   time.Time `json:"end_time"`
	Service   string    `json:"service,omitempty"`
	Level     string    `json:"level,omitempty"`
	TraceID   string    `json:"trace_id,omitempty"`
	Search    string    `json:"search,omitempty"`
	Limit     int       `json:"limit"`
	Offset    int       `json:"offset"`
}

type LogFilter struct {
	Field    string `json:"field"`
	Operator string `json:"operator"`
	Value    string `json:"value"`
}

type WebSocketMessage struct {
	Type    string      `json:"type"`
	Action  string      `json:"action,omitempty"`
	Data    interface{} `json:"data,omitempty"`
	Filters []LogFilter `json:"filters,omitempty"`
}

File: frontend/src/types/log.ts
export interface Log {
  id: string;
  timestamp: string;
  level: string;
  message: string;
  service: string;
  trace_id?: string;
  span_id?: string;
  attributes?: Record<string, any>;
}

export interface LogFilter {
  field: string;
  operator: string;
  value: string;
}

export interface WebSocketMessage {
  type: string;
  action?: string;
  data?: any;
  filters?: LogFilter[];
}

export type LogLevel = 'debug' | 'info' | 'warn' | 'error' | 'fatal';

export const LOG_LEVELS: LogLevel[] = ['debug', 'info', 'warn', 'error', 'fatal'];

export const LOG_LEVEL_COLORS: Record<LogLevel, string> = {
  debug: '#9e9e9e',
  info: '#2196f3',
  warn: '#ff9800',
  error: '#f44336',
  fatal: '#d32f2f',
};